File: solutions/java/logs-logs-logs/1/src/main/java/LogLevel.java
public enum LogLevel {
    TRACE(1),
    DEBUG(2),
    INFO(4),
    WARNING(5),
    ERROR(6),
    FATAL(42),
    UNKNOWN(0);

    private final int level;
    private LogLevel(int level){
        this.level = level;
    }

    public int getId(){
        return level;
    }

}

File: solutions/elixir/log-level/1/lib/log_level.ex
defmodule LogLevel do
  def to_label(level, legacy?) do
    case level do
      0 -> if legacy?, do: :unknown, else:  :trace
      1 -> :debug
      2 -> :info
      3 -> :warning
      4 -> :error
      5 -> if legacy?, do: :unknown, else: :fatal
      _ -> :unknown
    end
  end

  def alert_recipient(level, legacy?) do
    label = to_label(level, legacy?)

    case {label, legacy?} do
      {:error, _} -> :ops 
      {:fatal, _} -> :ops
      {:unknown, true} -> :dev1
      {:unknown, false} -> :dev2
      _ -> false
    end
  end
end
